ckanext/validation/plugin.py

```diff
@@ -188,6 +186,10 @@ def after_create(self, context, data_dict):
             for resource in data_dict.get(u'resources', []):
                 self._handle_validation_for_resource(context, resource)
         else:
+            for plugin in p.PluginImplementations(IDataValidation):
+                if not plugin.can_validate(context, data_dict):
+                    log.debug('Skipping validation for resource {}'.format(data_dict['id']))
+                    return
```

Comment:
This code checks for plugins implementing IDataValidation.  ckanext-unaids does this, and simply checks that the resourrce has the field "schema" in it, before continuing with validation. This was missing previously, meaning that whenever the logic landed here, validation was being triggered irrespective of whether the resource has a schema defined or not.